app/routines/find_or_create_user.rb

```diff
@@ -18,19 +18,33 @@ class FindOrCreateUnclaimedUser
   protected
 
   def exec(options)
-    unless options[:email] || options[:username]
-      fatal_error(code: :invalid_input, message: (I18n.t :"routines.find_or_create_unclaimed_user.must_provide_email_or_username"))
-    end
-
-    user = find_user(options)
-
     # output either the found unclaimed user or a freshly created one
-    outputs[:user] = user || create_user(options)
+    outputs.user = find_user(options) || create_user(options)
+  end
+
+  # Attempt to find a user by either the external_id, username, or email address
+  def find_user(options)
+    if options[:external_id].present?
+      User.joins(:external_ids).find_by(external_ids: { external_id: options[:external_id] })
+    elsif options[:username].present?
```

Comment:
are you aware of any applications still using username?

Reply:
No, not that I know of